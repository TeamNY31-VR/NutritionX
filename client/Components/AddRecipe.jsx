import {
  FormControl,
  InputLabel,
  Input,
  TextField,
  FormHelperText,
  FormLabel,
  Typography,
  Button,
  Card,
  Grid,
} from '@mui/material';
import { Box } from '@mui/system';
import React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { recipeActions } from '../store/recipes-slice';
import { addRecipe, syncRecipes, showFavoriteRecipes } from '../store/recipes-slice';

const AddRecipe = () => {
  const dispatch = useDispatch();
  const newRecipeName = useSelector((state) => state.recipes.newRecipeName);
  const newIngredientsList = useSelector(
    (state) => state.recipes.newIngredientsList
  );

  const recipeNameHandler = (value) => {
    dispatch(recipeActions.setRecipeName(value));
  };
  const ingredientsListHandler = (value) => {
    dispatch(recipeActions.setIngredientList(value));
  };

  const body = {
    name: newRecipeName,
    query: newIngredientsList,
  };

  const clearFields = () => {
    dispatch(recipeActions.setRecipeName(''));
    dispatch(recipeActions.setIngredientList(''));
  };

  return (
    <Card id="AddRecipe" variant="outlined" sx={{ p: 2 }}>
      <FormControl sx={{ width: '100%' }}>
        <Box container sx={{ display: 'flex' }}>
          <Typography variant="h4" aria-level="4">
            Add Your Meal
          </Typography>
          <TextField
            required
            label="Name"
            // id="fullWidth"
            sx={{ mx: 2 }}
            value={newRecipeName}
            onChange={(e) => recipeNameHandler(e.target.value)}
          />
          <Box sx={{ mr: 2, flexGrow: 1 }}>
            <TextField
              // fullWidth
              required
              label="Ingredients"
              // id="fullWidth"
              sx={{ width: '100%' }}
              // flex={1}
              // multiline={true}
              // rows={11}
              value={newIngredientsList}
              onChange={(e) => ingredientsListHandler(e.target.value)}
            />
          </Box>
          <Button
          id="add-btn"
            type="submit"
            variant="outlined"
            onClick={() =>
              dispatch(addRecipe(body))
                .then(() => dispatch(recipeActions.setRecipeName('')))
                .then(() => dispatch(recipeActions.setIngredientList('')))
                .then(() => dispatch(syncRecipes()))
            }
          >
            Add
          </Button>
          <Button
          id="show-faves-btn"
            type="submit"
            variant="outlined"
            onClick={() =>
              dispatch(addRecipe(body)).then(() =>
                dispatch(showFavoriteRecipes())
              )
            }
          >
            Show Favorites
          </Button>
          <Button
            id="show-all-btn"
            type="submit"
            variant="outlined"
            onClick={() =>
              dispatch(addRecipe(body)).then(() => dispatch(syncRecipes()))
            }
          >
            Show All
          </Button>
        </Box>
      </FormControl>
    </Card>
  );
};

export default AddRecipe;
