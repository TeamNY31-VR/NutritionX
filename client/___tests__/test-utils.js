import React from 'react';
import { render as rtlRender} from '@testing-library/react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import recipeReducer from '../store/recipes-slice';
import { initialRecipesState } from '../store/recipes-slice';

const render = (
  ui, {
    initialState = initialRecipesState,
    store = createStore(recipeReducer, initialState), 
    ...renderOptions
  } = {}
) => {
  const Wrapper = ({ children }) => (
    <Provider store={store}>{children}</Provider>
  );
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions});
};

export * from '@testing-library/react';
export { render }
