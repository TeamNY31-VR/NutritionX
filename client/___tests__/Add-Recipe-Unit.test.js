import React from 'react';
import AddRecipe from '../Components/AddRecipe';
import { render, screen } from './test-utils';

describe('<AddRecipe /> component', () => {
  let addRecipeComponent;

  beforeEach(() => {addRecipeComponent = render(<AddRecipe />)});
 
  it('addRecipeComponent renders correctly',() => {
    expect(addRecipeComponent).toMatchSnapshot();
  });

  it('renders 3 buttons to page', () => {
    expect(screen.getAllByRole('button')).toHaveLength(3);
  })

  it('renders a heading to the page', () => {
    expect(screen.getByRole('heading')).toBeInTheDocument();
  })
  
  it('renders form input field name to the page', () => {
    expect(screen.getByRole('textbox', { name: 'Name' })).toBeInTheDocument();
  })
});