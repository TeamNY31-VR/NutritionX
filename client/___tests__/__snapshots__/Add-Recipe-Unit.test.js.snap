// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddRecipe /> component addRecipeComponent renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root css-13svwm0-MuiPaper-root-MuiCard-root"
        id="AddRecipe"
      >
        <div
          class="MuiFormControl-root css-q8hpuo-MuiFormControl-root"
        >
          <div
            class="MuiBox-root css-k008qs"
          >
            <h4
              aria-level="4"
              class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
            >
              Add Your Meal
            </h4>
            <div
              class="MuiFormControl-root MuiTextField-root css-1nig0ce-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for="mui-1"
                id="mui-1-label"
              >
                Name
                <span
                  aria-hidden="true"
                  class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                >
                   
                  *
                </span>
              </label>
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id="mui-1"
                  required=""
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-173wfxe"
                  >
                    <span>
                      Name
                       
                      *
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              class="MuiBox-root css-keapom"
            >
              <div
                class="MuiFormControl-root MuiTextField-root css-136h9lv-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for="mui-2"
                  id="mui-2-label"
                >
                  Ingredients
                  <span
                    aria-hidden="true"
                    class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                  >
                     
                    *
                  </span>
                </label>
                <div
                  class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                    id="mui-2"
                    required=""
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-173wfxe"
                    >
                      <span>
                        Ingredients
                         
                        *
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
            <button
              class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
              id="add-btn"
              tabindex="0"
              type="submit"
            >
              Add
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
              id="show-faves-btn"
              tabindex="0"
              type="submit"
            >
              Show Favorites
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
              id="show-all-btn"
              tabindex="0"
              type="submit"
            >
              Show All
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root css-13svwm0-MuiPaper-root-MuiCard-root"
      id="AddRecipe"
    >
      <div
        class="MuiFormControl-root css-q8hpuo-MuiFormControl-root"
      >
        <div
          class="MuiBox-root css-k008qs"
        >
          <h4
            aria-level="4"
            class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
          >
            Add Your Meal
          </h4>
          <div
            class="MuiFormControl-root MuiTextField-root css-1nig0ce-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for="mui-1"
              id="mui-1-label"
            >
              Name
              <span
                aria-hidden="true"
                class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id="mui-1"
                required=""
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-173wfxe"
                >
                  <span>
                    Name
                     
                    *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiBox-root css-keapom"
          >
            <div
              class="MuiFormControl-root MuiTextField-root css-136h9lv-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for="mui-2"
                id="mui-2-label"
              >
                Ingredients
                <span
                  aria-hidden="true"
                  class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                >
                   
                  *
                </span>
              </label>
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id="mui-2"
                  required=""
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-173wfxe"
                  >
                    <span>
                      Ingredients
                       
                      *
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <button
            class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
            id="add-btn"
            tabindex="0"
            type="submit"
          >
            Add
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
            id="show-faves-btn"
            tabindex="0"
            type="submit"
          >
            Show Favorites
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
            id="show-all-btn"
            tabindex="0"
            type="submit"
          >
            Show All
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
